bootstrap/service.go

```diff
@@ -262,26 +286,47 @@ func (bs bootstrapService) UpdateConnections(ctx context.Context, token, id stri
 			return ErrThings
 		}
 	}
-	if err := bs.configs.UpdateConnections(ctx, owner, id, channels, connections); err != nil {
+	if err := bs.configs.UpdateConnections(ctx, user.GetDomainId(), id, channels, connections); err != nil {
 		return errors.Wrap(errUpdateConnections, err)
 	}
 	return nil
 }
 
 func (bs bootstrapService) List(ctx context.Context, token string, filter Filter, offset, limit uint64) (ConfigsPage, error) {
-	owner, err := bs.identify(ctx, token)
+	user, err := bs.identify(ctx, token)
 	if err != nil {
 		return ConfigsPage{}, errors.Wrap(svcerr.ErrAuthentication, err)
 	}
-	return bs.configs.RetrieveAll(ctx, owner, filter, offset, limit), nil
+
+	configsPage := bs.configs.RetrieveAll(ctx, user.GetDomainId(), filter, offset, limit)
+
+	j := 0
+	for _, config := range configsPage.Configs {
+		_, err := bs.authorize(ctx, user.GetDomainId(), auth.UsersKind, user.GetId(), auth.ViewPermission, auth.ThingType, config.ThingID)
```

Comment:
This doesn't seem good since we will make n calls to auth depending on the list of configs. You can check something like this https://github.com/absmach/magistrala/pull/2168. So if they aren't domain admins they get basic information. @arvindh123 what do you think?

Reply:
Yes, This should be like, first we need to get all `thingID`s from spiceDB for user and then pass the `thingID`s to config repo to retrieve the bootstrap config